src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1841,6 +1841,37 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - Determines if the given URI is currently supported
+    // Arguments:
+    // - The parsed URI
+    // Return value:
+    // - True if we support it, false otherwise
+    bool TerminalPage::_IsUriSupported(const winrt::Windows::Foundation::Uri& parsedUri)
+    {
+        if (parsedUri.SchemeName() == L"http" || parsedUri.SchemeName() == L"https")
+        {
+            return true;
+        }
+        if (parsedUri.SchemeName() == L"file")
+        {
+            const auto host = parsedUri.Host();
+            // If no hostname was provided or if the hostname was "localhost", Host() will return null
```

Comment:
In the comment it is written that the Host() will return null, yet the comparison is done with empty string

Unrelated to this, can 127.0.0.1 be supported as well?

Reply:
So WinRT is weird like that! The empty string _is_ treated as `null`, as strings aren’t pointer types across the ABI. It’s definitely unusual.